kernel/kernel-api/src/test/scala/io/delta/kernel/internal/icebergcompat/IcebergUniversalFormatMetadataValidatorAndUpdaterSuite.scala

```diff
@@ -0,0 +1,121 @@
+import io.delta.kernel.internal.TableConfig
+import io.delta.kernel.internal.actions.Metadata
+import io.delta.kernel.internal.util.ColumnMappingSuiteBase
+import io.delta.kernel.types.IntegerType
+import io.delta.kernel.types.StructType
+
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+class IcebergUniversalFormatMetadataValidatorAndUpdaterSuite extends AnyFunSuiteLike
+    with ColumnMappingSuiteBase {
+  test("validateAndUpdate should return empty when UNIVERSAL_FORMAT_ENABLED_FORMATS is not set") {
+    val metadata = createMetadata(Map("unrelated_key" -> "unrelated_value"))
+    val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+      createMetadata(),
+      metadata)
+    assert(!result.isPresent)
+  }
+
+  test(
+    "validateAndUpdate should return empty when no iceberg in UNIVERSAL_FORMAT_ENABLED_FORMATS") {
+    val metadata = createMetadata(Map(
+      TableConfig.UNIVERSAL_FORMAT_ENABLED_FORMATS.getKey -> "hudi",
+      "unrelated_key" -> "unrelated_value"))
+    val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+      createMetadata(),
+      metadata)
+    assert(!result.isPresent)
+  }
+
+  test("validateAndUpdate should return empty when iceberg is enabled" +
+    " and ICEBERG_COMPAT_V2_ENABLED is true") {
+    val metadata = createMetadata(Map(
+      TableConfig.UNIVERSAL_FORMAT_ENABLED_FORMATS.getKey -> "iceberg,hudi",
+      TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true",
+      "unrelated_key" -> "unrelated_value"))
+    val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+      createMetadata(),
+      metadata)
+    assert(!result.isPresent)
+  }
+
+  Seq(
+    Map(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "false"),
+    Map[String, String]()).foreach { disabledIcebergCompatV2Enabled =>
+    test(s"validateAndUpdate should remove iceberg when" +
+      s" ICEBERG_COMPAT_V2_ENABLED $disabledIcebergCompatV2Enabled") {
+      val metadata = createMetadata(Map(
+        TableConfig.UNIVERSAL_FORMAT_ENABLED_FORMATS.getKey -> "iceberg,hudi",
+        "unrelated_key" -> "unrelated_value") ++ disabledIcebergCompatV2Enabled)
+      val result = IcebergUniversalFormatMetadataValidatorAndUpdater.validateAndUpdate(
+        createMetadata(Map(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true")),
+        metadata)
+      assert(result.isPresent)
+      assert(TableConfig.UNIVERSAL_FORMAT_ENABLED_FORMATS.fromMetadata(result.get()) == Set(
+        "hudi").asJava)
+
+      val updatedConfig = result.get().getConfiguration
+      assert(updatedConfig.get("unrelated_key") === "unrelated_value")
+    }
+  }
+
+  Seq(
+    Map(TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "false"),
+    Map[String, String]()).foreach { disableIcebergCompatV2Enabled =>
+    test("validateAndUpdate should remove key entirely when only iceberg is present " +
+      s"and not compatible $disableIcebergCompatV2Enabled") {
```

Comment:
wehen icebergCompatV2 is disabled?

Reply:
I think I addressed the comment here, let me know if you would still like to rephrase.